src/macaron/slsa_analyzer/provenance/expectations/expectation.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
```

Comment:
yes, it was.

Reply:
>I believe this align with the goal that the Souffle policy should only be called from the policy engine entrypoint rather than having it as a "Policy" that will be run e2e?

Yes that's right.

>I wonder how much of it we have addressed or are these TODOs not relevant anymore

The TODOs that have been addressed have been removed